modules/light-clients/08-wasm/types/types_test.go

```diff
@@ -71,7 +77,41 @@ func GetSimApp(chain *ibctesting.TestChain) *simapp.SimApp {
 func setupTestingApp() (ibctesting.TestingApp, map[string]json.RawMessage) {
 	db := dbm.NewMemDB()
 	encCdc := simapp.MakeTestEncodingConfig()
-	app := simapp.NewSimApp(log.NewNopLogger(), db, nil, true, simtestutil.EmptyAppOptions{})
+	app := simapp.NewSimApp(log.NewNopLogger(), db, nil, true, simtestutil.EmptyAppOptions{}, nil)
+	return app, simapp.NewDefaultGenesisState(encCdc.Codec)
+}
+
+// SetupWasmTendermint sets up 2 chains and stores the tendermint/cometbft light client wasm contract on both.
+func (suite *TypesTestSuite) SetupWasmWithMockVM() {
+	ibctesting.DefaultTestingAppInit = suite.setupWasmWithMockVM
+
+	suite.coordinator = ibctesting.NewCoordinator(suite.T(), 1)
+	suite.chainA = suite.coordinator.GetChain(ibctesting.GetChainID(1))
+	suite.chainA.SetWasm(true)
+	suite.coordinator.SetCodeHash(suite.codeHash)
+}
+
+func (suite *TypesTestSuite) setupWasmWithMockVM() (ibctesting.TestingApp, map[string]json.RawMessage) {
+	suite.mockVM = &wasmtesting.MockWasmEngine{}
+	// TODO: move default functionality required for wasm client testing to the mock VM
```

Comment:
I think it'd be easier to address this in a followup when we better define how the mock VM will be used

Reply:
aye, I'll be adding some funcs here since we'd need to define ones for storing code and pinning (relevant to issue #4849). I'll open another issue to track this TODO.